### ISSUE TITLE
neovim

### ISSUE THREAD

**Issue by vhakulinen** [ervandew/eclim#385]
Is there any plans to make eclim work with neovim? (and with async support)

**Comment by ervandew** [comment]
Not currently, but once it looks like it has stabilized for a bit, I was planning on taking some time to evaluate it. One thing I do want to avoid is having to maintain integrations for multiple editors. I'd rather just devote my time towards the best one.

**Comment by dani-h** [comment]
How is the evaluation going?

**Comment by rkannan82** [comment]
I am going to take a look at what it takes to get it working with neovim. I will update after couple of weeks. I will be using this branch to track the changes: https://github.com/rkannan82/eclim/tree/neovim

**Comment by jasper-lyons** [comment]
I've been running eclim on neovim with no problems for a month now.

(Though obviously it blocks the UI sometimes, as with vim)

**Comment by rkannan82** [comment]
Yup. It works fine without any issues. The goal is to use neovim capabilities to do async processing for time consuming operations such as building the project, updating the window with error/warning marks, etc.

**Comment by phcerdan** [comment]
How is this going? The async build with neovim is gonna be amazing for Eclim. I use the CDT plugin (c++) and I love it, but usually I have to stop eclim because the freezing at build time.

**Comment by vito-c** [comment]
+1

**Comment by juanes852** [comment]
Please, how do you use Eclim in NeoVim. I'm on Arch Linux, and with the same Vimrc Eclim works fine with Vim, but no with Neovim. Eclim is not loaded in Neovim, and when I write « :PingEclim » in Neovim, I have « E492: Not an editor command: PingEclim ».

**Comment by justinmk** [comment]
@juanes852 Just a sanity check, did you follow https://neovim.io/doc/user/vim_diff.html#nvim-configuration ?

**Comment by juanes852** [comment]
@justinmk  I followed, https://neovim.io/doc/user/nvim_from_vim.html : ~/.nvimrc and ~/.nvim are symbolic links, and I use some « if [!]has('nvim') » to use Deoplete in Neovim and Neocomplete in Vim. Do you have specific configurations to use Eclim in Neovim ?

**Comment by phcerdan** [comment]
@juanes852 Eclim works out the box in neovim for me. Might be unrelated with your problem but if anyone is having problems setting Eclim up within neocompleteme, someone posted a workaround here: https://www.reddit.com/r/vim/comments/30phfv/neocomplete_and_eclim_not_working_well_together/

**Comment by juanes852** [comment]
@phcerdan, thanks, but is not the problem because when I use a minimalistic .vimrc, I have the same error. 

I think I've found the bug. When you install Eclim with « Web Developper Tools Support », Eclim doesn't work with Neovim, Eclim is not detected. Did you have the same problem ? With only Java support, Eclim works well for me in Neovim.

**Comment by juanes852** [comment]
Again, Eclim no longer has been recognized by Neovim. I've found a best solution to use Eclim in Neovim. I've found https://github.com/starcraftman/vim-eclim, I've installed with NeoBundle, and with a simple adatation Eclim works very well in Neovim ! To have the last one Eclim, I made a symlink in ~/.vim/bundle/vim-eclim to the target ~/.nvim/eclim and it works out of the box (do not made in step 2 of starcraftman/vim-eclim).

**Comment by JulioJu** [comment]
I've seen there is a problem with interactive Java programs, and Java programms with infinite loops. We can see on https://github.com/neovim/neovim/wiki/Troubleshooting  « Interactive commands are supported by the :terminal command in Neovim. However,  :! and system() do not support interactive commands ».

Please, could you implement a solution to send output of command `Java %` not in the Quickfix but in the [Terminal](https://neovim.io/doc/user/nvim_terminal_emulator.html) in Neovim `if has('nvim') […] endif ` ?

I think, this new command `Java %`» for Neovim could use asynchronous plugin [Neomake](https://github.com/benekastah/neomake) with eclim as  « makers » ? 

Thanks in advance

**Comment by tnguyen14** [comment]
Does anyone know how I can install it to work with neovim? I've run the installer once, and it's working in vim for me, but there's no `PingEclim` command in neovim.

**Comment by JulioJu** [comment]
@tnguyen14  just sanity check, have you follow :help nvim-from-vim ?

**Comment by tnguyen14** [comment]
@JulioJu thanks I'll try to symlink `~/.vim` to `~/.config/nvim`.

**Comment by ttiurani** [comment]
Async support for neovim/neomake would be so useful to me now, that I might find time to work on this. Are there any preliminary ideas anywhere @ervandew as to how this should be implemented?

**Comment by mvalkon** [comment]
@ttiurani async support would be _so_ nice. 👍 if you are going to do it

**Comment by Cypher1** [comment]
Wondering how this is going and what needs to be done to get eclim with neovim working now.
I have had some trouble installing it and have decided (until I learn more) that I should just use Ant builds and artur-shaik/vim-javacomplete2 (with deoplete) for autocompleting.

Thanks

**Comment by vhakulinen** [comment]
@Cypher1 and everyone else:

FYI: Language Server[1] might change the playground quite a bit. IIRC there even are some plans to support it out-of-the-box in neovim. Also iirc vscode uses that system.

[1]: https://github.com/Microsoft/language-server-protocol

**Comment by componhead** [comment]
Hi. Eclim seems doesn't work with neovim; there is a planning about compatibility?

thank you

**Comment by JulioJu** [comment]
@componhead Eclim works perfectly with Neovim. Have you seen `:help nvim-from-vim` ?

**Comment by componhead** [comment]
Mmmh. Yes I did.

I have some problems with installation. This is my stack flow
`
`

**Comment by JulioJu** [comment]
@componhead , in your file `eclim-issue.txt`, at line 8, we could read "`java.io.IOException: Cannot run program "gvim": error=2, No such file or directory`"

Probably eclim neads gvim to be installed. So, try to install it (`apt-get install gvim`).

**Comment by componhead** [comment]
I installed `vim-gtk3` but without luck (gvim doesn't exist anymore on my sources).

It crash building too.
[eclim-building.txt](https://github.com/ervandew/eclim/files/1421190/eclim-building.txt)

**Comment by JulioJu** [comment]
@componhead. With Eclipse Oxygen, you must use Eclim 2.7.0 and not Eclim 2.6.0 as you have done (see lines 23 and 24 of your file). See http://eclim.org/install.html.
